exporter.go

```diff
@@ -102,6 +127,29 @@ func (e *Exporter) Export(
 			}
 			bpMD.Layers[layer.name()] = lmd
 		}
+		meta.Buildpacks = append(meta.Buildpacks, bpMD)
+
+		// Process cached layers.
+		bpCacheMD := metadata.BuildpackLayersMetadata{
+			ID:      bp.ID,
+			Version: bp.Version,
+			Layers:  map[string]metadata.BuildpackLayerMetadata{},
+		}
+		for _, layer := range bpDir.findLayers(cached) {
+			if !layer.hasLocalContents() {
+				return fmt.Errorf("failed to cache layer %q because it has no contents", layer.Identifier())
+			}
+			lmd, err := layer.read()
+			if err != nil {
+				return errors.Wrapf(err, "reading %q metadata", layer.Identifier())
+			}
+			origLayerMetadata := origCacheMetadata.MetadataForBuildpack(bp.ID).Layers[layer.name()]
+			if lmd.SHA, err = e.addOrReuseCacheLayer(cacheStore, &layer, origLayerMetadata.SHA); err != nil {
+				return err
+			}
+			bpCacheMD.Layers[layer.name()] = lmd
```

Comment:
Is it better to hide the map and access it with methods?

Reply:
I considered that, but since there is no logic involved it seems to make the logic harder to follow since you now have to go to another file to see what the function does. Creating the toml/json objects by hand seems to be a repeating pattern in lifecycle.